components/script/dom/bindings/global.rs

```diff
@@ -143,6 +143,15 @@ impl<'a> GlobalRef<'a> {
         }
     }
 
+    /// Get the [base url](https://html.spec.whatwg.org/multipage/#api-base-url)
+    /// for this global scope.
+    pub fn api_base_url(&self) -> Url {
+        match *self {
+            GlobalRef::Window(ref window) => window.Document().base_url(),
+            GlobalRef::Worker(ref worker) => worker.get_url().clone(),
```

Comment:
Link to <https://html.spec.whatwg.org/multipage/#script-settings-for-browsing-contexts:api-base-url> and <https://html.spec.whatwg.org/multipage/#script-settings-for-workers:api-base-url> above the relevant arm.